src/compiler/checker.ts

```diff
@@ -7154,35 +7154,36 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     const methodDeclaration = signatureToSignatureDeclarationHelper(signature, SyntaxKind.MethodSignature, context, { name: propertyName, questionToken: optionalToken }) as MethodSignature;
                     typeElements.push(preserveCommentsOn(methodDeclaration));
                 }
+                if (signatures.length || !optionalToken) {

```

Comment:
I think canonically we write `some(signatures)` but that's a nit

Reply:
That's perhaps somewhat useful if `signatures` could be `undefined` - but here it can't so I'm not super inclined to change this as the current form feels more idiomatic to me. I don't mind to change it if it gets requested more "firmly" though ;p